libcxx/test/std/utilities/optional/optional.iterator/iterator.pass.cpp

```diff
@@ -19,6 +19,21 @@
 #include <type_traits>
 #include <utility>
 
+template <typename T>
+concept has_iterator = requires { typename T::iterator; };
+
+template <typename T>
+concept has_const_iterator = requires { typename T::const_iterator; };
+
+template <typename T>
+concept has_both_iterators = has_iterator<T> && has_const_iterator<T>;
+
+template <typename T>
+concept only_has_iterator = has_iterator<T> && !has_const_iterator<T>;
+
+template <typename T>
+concept has_no_iterators = !has_iterator<T> && !has_const_iterator<T>;
+
 template <typename T>
 constexpr bool test_range_concept() {
   return std::ranges::range<std::optional<T>>;
```

Comment:
It seems more senseful to make this a `concept` or `constexpr bool` variable template.